docs/docs/0.3.2/policies/proxy-template.md

```diff
@@ -1,38 +1,140 @@
+
+```yaml
+type: ProxyTemplate
+mesh: default
+name: my-custom-template
+
+# `selectors` define a subset of `Dataplanes`
+# this `ProxyTemplate` should apply to
+selectors:
+  # selector matches tags either on an `inbound` or
+  # on a `gateway` interface of a `Dataplane`
+  - match:
+      # the following selector will match a very specific
+      # subset of `Dataplanes` in the `Mesh`
+      service: backend
+      role:    api
+
+# `conf` defines the desired dataplane (Envoy) configuration
+conf:
+  # `imports` let you reuse dataplane configuration that Kuma
+  # can generate automatically and add a few tweaks on top of it
+  imports:
+    # `default-proxy` is a reference name for the default
+    # dataplane configuration generated by Kuma
+    - default-proxy
+
+  # `resources` define a list of raw Envoy resources
+  # that will either complement or override auto-generated ones
+  resources:
+    - name: localhost:9901
+      version: v1
+      # `resource` is a raw Envoy resource
+      # provided either in YAML or in JSON format
+      resource: |
+        '@type': type.googleapis.com/envoy.api.v2.Cluster
+        connectTimeout: 5s
+        name: localhost:9901
+        ...
+```
+
+Notice that:
+
+1. `selectors` allow you to limit the scope of [Dataplanes](../documentation/dps-and-data-model/#dataplane-entity) this `ProxyTemplate` should apply to
+2. `imports` allow you to reuse configuration that Kuma can generate automatically and add a few tweaks on top of it
+3. `resources` allow you to provide raw Envoy resources that will either complement or override auto-generated ones
+
+::: tip
+In the current release, the only available canned configuration that can be used inside `imports` section is called `default-proxy`.
+
+In future releases, more of these will be available and it will also be possible for a user to define them to re-use across their infrastructure.
+:::
+
+At runtime, whenever the `Kuma Control Plane` needs to generate configuration for a given [Dataplane](../documentation/dps-and-data-model/#dataplane-entity), it will proceed as follows:
+* First, it will search for all `ProxyTemplates` that have been defined in the [Mesh](./mesh/) a given [Dataplane](../documentation/dps-and-data-model/#dataplane-entity) belongs to
+* Next, it will select only those `ProxyTemplates` whose `selectors` [match](./how-kuma-chooses-the-right-policy-to-apply/) either an `inbound` or a `gateway` interface of a given [Dataplane](../documentation/dps-and-data-model/#dataplane-entity)
+* Next, every matching `ProxyTemplate` will be [ranked](./how-kuma-chooses-the-right-policy-to-apply/)
+* Then, `ProxyTemplate` with the highest rank will be used to generate configuration for that dataplane (Envoy)
+* If a `ProxyTemplate` defines `imports`, their resources will be generated first
+* If a `ProxyTemplate` defines `resources`, they will be copied "as is" and override auto-generated resources of the same name
```

Comment:
Questions that user may have:

Can I patch config generated by Kuma?
What if there is a conflict between resources (same name)?

Reply:
* switched to using word `replace` instead of `override`
* added a tip that explicitly says that PATH and DELETE are not supported